pkg/tfgen/docs.go

```diff
@@ -219,7 +228,29 @@ func getDocsBaseURL(org, p string) string {
 
 // getDocsDetailsURL gets the detailed resource or data source documentation source.
 func getDocsDetailsURL(org, p, kind, name string) string {
-	return fmt.Sprintf(docsDetailsURL, org, p, kind, name)
+	repoPath, err := getRepoPath(org, p)
+	if err != nil {
+		return ""
+	}
+
+	possibleMarkdownNames := []string{
+		name + ".html.markdown",
+		name + ".markdown",
+		name + ".html.md",
+	}
```

Comment:
Can we somehow unify this with the very similar logic we have for `possibleMarkdownNames` in this file?  It feels like there should be some common logic for finding the right names that we use across both of there (and other places as well - like the suspicious looking code in `RenameDataSource`).

Reply:
Sure! I will try and refactor this more - I just didn't want to mess with any of the existing logic :)